src/main/java/me/karakelley/http/controllers/Router.java

```diff
@@ -16,6 +18,10 @@ public Router route(String path, Controller controller) {
 
   @Override
   public Response respond(Request request) {
-    return mappings.getOrDefault(request.getPath(), new InvalidRequestController()).respond(request);
+    if (mappings.containsKey(request.getPath())) {
+      return mappings.get(request.getPath()).respond(request);
+    } else {
+      return mappings.getOrDefault(request.getPath(), new InvalidRequestController()).respond(request);
```

Comment:
Why `getOrDefault` in the `else` branch. Won't it always be the default?

Reply:
It always defaults if there is a directory passed in when the server starts. But, If there is no directory and you try to hit a route that hasn't been predefined (i.e. `/` or `/redirectme` then it will default.

Reply:
I've thought about this for like 5 minutes now and I still don't understand why you can't replace lines 21-25 with just...

```
return mappings.getOrDefault(request.getPath(), new InvalidRequestController()).respond(request);
```

Reply:
Yeah,  you're right. For some reason I kept trying to remove line 24 instead of 21-23

Reply:
👍